File: merge-sort/merge_sort.php
<?php
function mergeSort($arr) {
    if (count($arr) <= 1) return $arr;
    
    $mid = intval(count($arr) / 2);
    $left = mergeSort(array_slice($arr, 0, $mid));
    $right = mergeSort(array_slice($arr, $mid));
    
    return merge($left, $right);
}

function merge($left, $right) {
    $result = [];
    $i = $j = 0;
    
    while ($i < count($left) && $j < count($right)) {
        if ($left[$i] <= $right[$j]) {
            $result[] = $left[$i++];
        } else {
            $result[] = $right[$j++];
        }
    }
    
    while ($i < count($left)) $result[] = $left[$i++];
    while ($j < count($right)) $result[] = $right[$j++];
    
    return $result;
}

$arr = [64, 34, 25, 12, 22, 11, 90];
print_r(mergeSort($arr));
?>

File: merge-sort/merge_sort.clj
(defn merge-arrays [left right]
  (loop [result [] l left r right]
    (cond
      (empty? l) (concat result r)
      (empty? r) (concat result l)
      (<= (first l) (first r)) (recur (conj result (first l)) (rest l) r)
      :else (recur (conj result (first r)) l (rest r)))))

(defn merge-sort [arr]
  (if (<= (count arr) 1)
    arr
    (let [mid (quot (count arr) 2)
          left (merge-sort (take mid arr))
          right (merge-sort (drop mid arr))]
      (merge-arrays left right))))

(println (merge-sort [64 34 25 12 22 11 90]))
